src/app/pages/common/entity/entity-form/components/form-chip/form-chip.component.ts

```diff
@@ -67,27 +69,47 @@ export class FormChipComponent implements Field, OnInit {
   }
 
   selected(event: MatAutocompleteSelectedEvent): void {
-    this.chipLists.push(event.option.viewValue);
+    if (this.config.selectOnly) {
+      if (this.chipLists.includes(event.option.viewValue)) {
+        this.chipLists = this.chipLists.filter((e) => e !== event.option.viewValue);
+      } else {
+        this.chipLists = [...new Set([...this.chipLists, event.option.viewValue])];
+      }
+    } else {
+      this.chipLists.push(event.option.viewValue);
+    }
+    this.group.controls[this.config.name].setValue(this.chipLists);
     this.chipInput.nativeElement.value = '';
     this.chipCtrl.setValue(null);
+    this.openPanel();
   }
 
   updateSearchOptions(value: any): void {
     if (this.config.updater && this.config.parent) {
+      const values = this.chipLists as string[];
       if (this.config.updateLocal) {
-        this.config.updater(value, this.config.parent, this.config);
+        this.config.updater(value, values, this.config.parent, this.config);
       } else {
-        this.config.updater(value, this.config.parent);
+        this.config.updater(value, values, this.config.parent);
       }
     } else {
       value = value.toLowerCase();
       const searchOptions: Option[] = [];
       this.config.options.forEach((option) => {
-        if (option.label.toLowerCase().includes(value)) {
+        if (!value || option.label.toLowerCase().includes(value)) {
           searchOptions.push(option);
         }
       });
       this.config.searchOptions = searchOptions;
     }
   }
+
+  openPanel(): void {
+    if (this.config.selectOnly) {
+      setTimeout(() => {
+        this.trigger._onChange('');
```

Comment:
This `_onChange` doesn't seem to have any effect. Furthermore, it's a private method. We shouldn't be using it. Maybe it can be replaced with `this.trigger.writeValue('')`. I'm not sure. But as is, it doesn't seem to be doing anything. Please elaborate on its purpose here.

Reply:
Right. Shouldn't change it manually.